tasks/install.yml

```diff
@@ -52,14 +52,11 @@
   become: true
   when: not vault_configuration.disable_mlock
 
-- name: Generate systemd file
-  template:
-    src: vault.service.j2
-    dest: "/etc/systemd/system/vault.service"
-    owner: "{{ vault_user }}"
-    group: "{{ vault_group }}"
-    mode: 0644
-  become: true
+- include: setup-RedHat-6.yml
+  when: ansible_distribution_major_version == '6'
+
+- include: setup-RedHat-7.yml
+  when: ansible_distribution_major_version == '7'
 
```

Comment:
@christophershoemaker what about doing something like this:

```
- include: setup-RedHat-6.yml
  when: (ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux') and ansible_distribution_major_version|int == 6

- include: setup-RedHat-7.yml
  when: not ((ansible_distribution == 'CentOS' or ansible_distribution == 'Red Hat Enterprise Linux') and ansible_distribution_major_version|int == 6)
```

Basically make `setup-RedHat-7.yml` run when the system is *not* CentOS 6? Otherwise, we may have to have a very long conditional to account for all the variances.